proto/internalapi/scanner/v4/indexer_service.proto

```diff
@@ -33,7 +33,8 @@ message HasIndexReportRequest {
 }
 
 message HasIndexReportResponse {
-    bool exists = 1;
+    string scanner_version = 1;
+    bool   exists          = 2;
```

Comment:
Why do we have `exists` here? IIRC, the design of the APIs was to return status codes (i.e., `NotFound` gRPC error code) instead of false to communicate resources that do not exist in Scanner (it's confusing to clients (do they need to handle the status code, the boolean flag, or both?)

Reply:
I think I forgot about that original design and thought it was an oversight: https://github.com/stackrox/stackrox/pull/9210

That being said, if we want to show the scanner version for this endpoint, then I think adding the boolean makes sense. Perhaps we shouldn't bother with adding the version to this one, though?

Reply:
> That being said, if we want to show the scanner version for this endpoint, then I think adding the boolean makes sense. Perhaps we shouldn't bother with adding the version to this one, though?

Use unary interceptors on both server and client to add/read metadata headers named "scanner-version" from every request for future inspection.

So, before we settle, what is the use case for the version string? Debug? Checking version across Services? Validation? Knowing the use cases will give us a better idea of how to approach to this.

Reply:
Current usage is for https://github.com/stackrox/stackrox/pull/9342